src/swtpm/swtpm.c

```diff
@@ -231,11 +234,39 @@ int swtpm_main(int argc, char **argv, const char *prgname, const char *iface)
                         "Given file descriptor type is not supported.\n");
                 exit(1);
             }
-            mlp.flags |= MAIN_LOOP_FLAG_TERMINATE | MAIN_LOOP_FLAG_USE_FD;
+            mlp.flags |= MAIN_LOOP_FLAG_USE_FD |            /* Use this as client fd */
+                         MAIN_LOOP_FLAG_TERMINATE |         /* Terminate if requested */
+                         MAIN_LOOP_FLAG_KEEP_CONNECTION;    /* Keep connection till the client connection active */
             SWTPM_IO_SetSocketFD(mlp.fd);
 
             break;
 
+        case 'F':
+            val = strtoul(optarg, &end_ptr, 10);
+            if (val != (unsigned int)val || errno || end_ptr[0] != '\0') {
+                fprintf(stderr, "Cannot parse control channel socket file descriptor.\n");
+                exit(1);
+            }
+            mlp.ctrlfd = val;
+            if (fstat(mlp.ctrlfd, &statbuf) != 0) {
+                fprintf(stderr, "Cannot stat control channel file descriptor: %s\n", 
+                        strerror(errno));
+                exit(1);
+            }
+            /*
+             * test for wrong file types; anonymous fd's do not seem to be any of the wrong
+             * ones but are also not character devices
+             */
+            if (S_ISREG(statbuf.st_mode) || S_ISDIR(statbuf.st_mode) || S_ISBLK(statbuf.st_mode)
+                || S_ISLNK(statbuf.st_mode)) {
+                fprintf(stderr,
+                        "Given control channnel file descriptor type is not supported.\n");
+                exit(1);
+            }
+            mlp.flags |= MAIN_LOOP_FLAG_USE_CTRLFD;
+
+            break;
+
         case 'c':
             serverdata = optarg;
             break;
```

Comment:
I am not sure if i can provide a testcase, As tests are written using shell scripts - i have no idea how can we create and pass a unix socket to swtpm inside shell.